### ISSUE TITLE
JAWS doesn't read that something is a link when the link wraps certain elements

### ISSUE THREAD

**Issue by sivakusayan** [FreedomScientific/VFO-standards-support#642]
### Summary
When a link wraps an element with `display: table`, the link isn't announced as a link when focused.

1. Navigate to [this codepen](https://codepen.io/sivakusayan/pen/WNMdvvP).
2. Focus the link.
3. Notice that the link's name is read out, but no mentions that it is a link.

### Expected result
The link is announced as a link.

### Actual result
Just the link's name was announced, with no mention that it's a link.

### Additional Information
This works with NVDA. I also _think_ that this is valid HTML after reading [the documentation](https://html.spec.whatwg.org/#the-a-element) for HTML link specifications.

#### JAWS version and build number
JAWS Version 2022.2204.20 ILM
 
#### Operating System and version
Microsoft Windows 10 Enterprise
Version: 10.0.19042 Build 19042

#### Browser and version:
Chrome Version 101.0.4951.67 (Official Build) (64-bit)

**Comment by JAWS-test** [comment]
The problem also occurs when a link contains other elements, e.g. `<article>`:
* No output of the role "link" during navigation with the TAB key
* Wrong output of the order of the roles "article" and "link" when reading with the virtual cursor.
* In the link overview, in addition to the link label, "article ... end of article" is output
